uml-diagrams/FilterCoursesCommand.puml

```diff
@@ -0,0 +1,41 @@
+@startuml
+hide footbox
+
+actor CEG_Student
+participant ":FilterCoursesCommand" as FilterCoursesCommand
+participant ":Command" as Command
+
+CEG_Student -> FilterCoursesCommand: execute (userInput:String)
+activate FilterCoursesCommand
+
+FilterCoursesCommand -> Command: createJsonObject()
+activate Command
+
+opt inputStream is null
+    Command -> Command: throw new IOException()
+end
+
+Command --> FilterCoursesCommand: JsonObject
+deactivate Command
+
+FilterCoursesCommand -> FilterCoursesCommand: getNusCourseCode(userInput: String)
+activate FilterCoursesCommand
+return courseToFind : String
```

Comment:
`if (inputDetails.length == COMMAND_WORD_INDEX + ZERO_INDEX_OFFSET)` and `if (inputDetails.length > FILTER_COURSES_MAX_ARGS)`

Should we include this in the activation bar for `getNusCourseCode`?

Reply:
Good catch! These paths are to throw exceptions when the user has invalid input, so I am also not sure if I should put it into the diagram as I was thinking of showing the sequence when a valid input was given! What do you think? 🤔